pageserver/src/layered_repository.rs

```diff
@@ -1523,6 +1539,12 @@ impl LayeredTimeline {
                 &metadata,
                 false,
             )?;
+
+            // Update upload metrics
+            let new_delta_size = File::open(&new_delta_path)?.metadata()?.len();
```

Comment:
There is no need to open a file in order to get a metadata for it. `.metadata` is available on [PathBuf](https://doc.rust-lang.org/stable/std/path/struct.PathBuf.html#method.metadata)